client/handle_http.go

```diff
@@ -1069,6 +1073,45 @@ func (tc *TransferClient) NewTransferJob(ctx context.Context, remoteUrl *url.URL
 		}
 	}
 
+	// If we are a recursive download and using the director, we want to attempt to get directory listings from
+	// PROPFINDing the director
+	if recursive && !upload && tj.useDirector {
+		// Query the director a PROPFIND to see if we can get our directory listing
+		resp, err := queryDirector(tj.ctx, "PROPFIND", remoteUrl.Path, tj.directorUrl)
+		if err != nil {
+			return nil, err
+		}
+		// If the director responds with 405 (method not allowed), we're working with an old Director.
+		// In that event, we try to fallback and use the deprecated dirlisthost
+		if resp.StatusCode == http.StatusMethodNotAllowed {
+			log.Debugln("Failed to find collections url from director, attempting to find directory listings host in namespace")
+			// Check for a dir list host in namespace
+			if tj.namespace.DirListHost == "" {
+				// Both methods to get directory listings failed
+				return nil, errors.New("Origin and/or namespace does not support directory listings")
```

Comment:
Nitpick: make this a non-exported symbol at the module level (put it in the `var` block at the top of the file).  This way, we can do an `errors.Is` against it in the future.

Also, please follow go style and make error message not capitalized.